src/bin/ion/commands/beta/generate/templates/java/class.templ

```diff
@@ -135,43 +103,21 @@ public class {{ target_kind_name }} {
      * The caller is responsible for closing the stream associated with the writer.
      */
     public void writeTo(IonWriter writer) throws IOException {
-         {% if abstract_data_type == "Value" %}
-            {# Writes `Value` class with a single field `value` as an Ion value #}
-            {% for field in fields %}
-                {% if field.value_type | is_built_in_type == false  %}
-                    this.{{ field.name | camel }}.writeTo(writer)?;
-                {% else %}
-                    writer.write{{ field.isl_type_name | upper_camel }}(this.value);
-                {% endif %}
-            {% endfor %}
-         {% elif abstract_data_type is object and abstract_data_type is containing("Structure") %}
-             {# Writes `Structure` class with multiple fields based on `field.name` as an Ion struct #}
-             writer.stepIn(IonType.STRUCT);
-             {% for field in fields %}
-                 writer.setFieldName("{{ field.name }}");
-                 {% if field.value_type | is_built_in_type == false %}
-                     this.{{ field.name | camel }}.writeTo(writer);
+         {# Writes `Structure` class with multiple fields based on `field.name` as an Ion struct #}
+         writer.stepIn(IonType.STRUCT);
+         {% for field in fields %}
+             writer.setFieldName("{{ field.name }}");
+             {% if field.value_type | is_built_in_type == false %}
+                 {% if field.value_type is containing("ArrayList") %}
+                    {{ util_macros::write_as_sequence(field=field) }}
```

Comment:
(🗺️ PR tour) This statement is modified to write fields that are of type `ArrayList` using util_macros `write_as_sequence`. Previously it was not possible for a field in struct to be of sequence type(e.g. It would be inside some nested type `NestedtypeX`). But with this change it is possible to have fields with sequence type (e.g. `ArrayList<Inetger>`).